seed/seedwriter/writer_test.go

```diff
@@ -105,6 +106,15 @@ func (s *writerSuite) SetUpTest(c *C) {
 	s.db = db
 
 	retrieve := func(ref *asserts.Ref) (asserts.Assertion, error) {
+		// XXX: Having this support in 'retrieve' as a part of the Fetcher () allows
+		// us to very easily support fetching additional assertions that are related to the
+		// model.
+		if ref.Type.SequenceForming() {
+			_, seq, err := ref.SequenceKey()
+			if err == nil && seq <= 0 {
+				return ref.ResolveSequence(s.StoreSigning.FindSequence)
+			}
```

Comment:
this is a bit of  hack, I fear the right approach is to change the API of the Fetcher to add a method for Sequence forming assertions maybe taking an AtSequence, compare with asserts.Pool